rsciio/zspy/_api.py

```diff
@@ -102,17 +98,32 @@ def _get_object_dset(group, data, key, chunks, **kwds):
             chunks = data.shape
         these_kwds = kwds.copy()
         these_kwds.update(dict(dtype=object, exact=True, chunks=chunks))
-        test_ind = data.ndim * (0,)
-        # Need to know the underlying dtype for the codec
-        # Note this can't be an object array
-        if isinstance(data, da.Array):
-            dtype = data[test_ind].compute().dtype
+
+        if dtype is None:
+            test_data = data[data.ndim * (0,)]
```

Comment:
Can we just compute the data and then check the sub array type with the metadata like above? It seems slightly better than what we are doing now?

Reply:
Sorry, I am not sure if I follow what you are referring to.

Reply:
Anyways this looks good!

Reply:
> I just always feel like there should be a better way of doing this from a numpy object array but that doesn't seem like it is the case. We really want to be able to set a constant dtype for the ragged array. The current implementation is just too flexible for my liking.
> 
> That's a bit of an aside, I think if we really wanted to solve this problem we would have to add something like [awkwardarray](https://awkward-array.org/doc/main/) as a dependency.

Yes quite likely. For now, this fix should do, hopefully! ;)